programs/drift/src/state/order_params.rs

```diff
@@ -376,13 +376,40 @@ impl OrderParams {
         perp_market: &PerpMarket,
         direction: PositionDirection,
     ) -> DriftResult<i64> {
-        // price offsets baselines for perp market auctions
-        let mark_twap_slow = perp_market
+        if perp_market
             .amm
-            .last_ask_price_twap
-            .safe_add(perp_market.amm.last_bid_price_twap)?
-            .safe_div(2)?
-            .cast::<i64>()?;
+            .historical_oracle_data
+            .last_oracle_price_twap_ts
+            .safe_sub(perp_market.amm.last_mark_price_twap_ts)?
+            .abs()
+            >= 60
+        {
+            // if uncertain with timestamp mismatch, enforce within N bps
+            let price_divisor = if perp_market
+                .contract_tier
+                .is_as_safe_as_contract(&ContractTier::B)
+            {
+                500
+            } else {
+                100
+            };
+
+            return Ok(match direction {
+                PositionDirection::Long => {
+                    perp_market.amm.last_bid_price_twap.cast::<i64>()? / price_divisor
```

Comment:
might be good to add comment that it's starting at bid (beneath oracle) intentionally to get a good price?